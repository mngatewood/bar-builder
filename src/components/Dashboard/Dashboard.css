.dashboard-container {
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100vw;
  height: 50px;
  background-color: rgba(245, 244, 193, 1);
  z-index: 29;
}

.dashboard-container button {
  margin: 10px 25px;
  font-size: 18px;
  background-color: rgba(116, 121, 89, 1);
  font-weight: bold;
  color: rgba(245, 244, 193, 1);
  border-radius: 5px;
  box-shadow: 5px 5px 5px rgba(116, 121, 89, 1);
    text-shadow:
   -1px -1px 0 darkgreen,  
    1px -1px 0 darkgreen,
    -1px 1px 0 darkgreen,
     1px 1px 0 darkgreen;
}