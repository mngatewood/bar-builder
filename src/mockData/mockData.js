/* eslint-disable */
export const mockFetchResponse = () => {
  drinks: [
    { strDrink: "Apple Cider Punch #1", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/xrqxuv1454513218.jpg", idDrink: "12864" },
    { strDrink: "Chocolate Monkey", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/tyvpxt1468875212.jpg", idDrink: "15615" },
    { strDrink: "Egg Nog #4", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/wpspsy1468875747.jpg", idDrink: "12910" },
    { strDrink: "Egg Nog - Healthy", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/qxuppv1468875308.jpg", idDrink: "12916" },
    { strDrink: "Egg-Nog - Classic Cooked", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/quxsvt1468875505.jpg", idDrink: "12914" },
    { strDrink: "GG", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/vyxwut1468875960.jpg", idDrink: "15997" },
    { strDrink: "Gluehwein", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/vuxwvt1468875418.jpg", idDrink: "12944" },
    { strDrink: "Sunny Holiday Punch", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/rywtwy1468924758.jpg", idDrink: "13042" },
    { strDrink: "Wine Cooler", strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/yutxtv1473344210.jpg", idDrink: "13056" }
  ]
}