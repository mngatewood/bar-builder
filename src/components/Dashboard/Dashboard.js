import React, { Component } from 'react';
import './Dashboard.css';
import { NavLink, withRouter } from 'react-router-dom';
import { getMenuRecipes } from '../../api/apiHelpers/getMenuRecipes';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addMenuRecipes, clearMenuRecipes } from '../../actions';

export class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuRecipes: []
    }
  }

  handleMenuClick = async () => {
    const { inventory, ingredients, clearMenuRecipes, addMenuRecipes, history } = this.props
    await clearMenuRecipes();
    const menuRecipes = await getMenuRecipes(inventory, ingredients);
    // console.log(menuRecipes);
    await addMenuRecipes(menuRecipes);
    history.push('/menu');
  };

  render() {

    const { inventory, addMenuRecipes, clearMenuRecipes, history } = this.props

    return <aside className="dashboard-container">
      <NavLink to='/inventory'>
        <button>
          Update My Inventory
        </button>
      </NavLink>
        <button onClick={this.handleMenuClick}>
          View My Bar Menu
        </button>
    </aside>;

  }

};

export const mapStateToProps = state => ({
  inventory: state.inventory,
  ingredients: state.ingredients
});

export const mapDispatchToProps = dispatch => ({
  addMenuRecipes: recipes => dispatch(addMenuRecipes(recipes)),
  clearMenuRecipes: () => dispatch(clearMenuRecipes())
});

Dashboard.propTypes = {
  inventory: PropTypes.array
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Dashboard));