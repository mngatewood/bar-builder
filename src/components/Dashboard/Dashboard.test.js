import { Dashboard, mapStateToProps, mapDispatchToProps } from './Dashboard';
import { shallow } from 'enzyme';
import React from 'react';
import * as mock from '../../mockData/mockData';

describe("Dashboard", () => {

  it('matches the snapshot', () => {
    const addMenuRecipes = jest.fn();
    const clearMenuRecipes = jest.fn();
    const wrapper = shallow(<Dashboard
      addMenuRecipes={addMenuRecipes}
      clearMenuRecipes={clearMenuRecipes}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('maps to the store', () => {
    const mockStore = { 
      inventory: mock.mockInventory,
      ingredients: mock.sortedIngredientOptions };
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  });

  it('calls dispatch function when using mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addMenuRecipes();
    mapped.clearMenuRecipes();
    expect(mockDispatch).toHaveBeenCalled();
  });

});

describe("handleMenuClick", () => {

  it.skip('calls clearMenuRecipes, getMenuRecipes, addMenuRecipes with expected params', () => {
    const addMenuRecipes = jest.fn();
    const clearMenuRecipes = jest.fn();
    const getMenuRecipes = jest.fn();
    const wrapper = shallow(<Dashboard
      addMenuRecipes={addMenuRecipes}
      clearMenuRecipes={clearMenuRecipes}
      getMenuRecipes={getMenuRecipes} />);

    wrapper.instance().handleMenuClick();
    expect(clearMenuRecipes).toHaveBeenCalled;
    expect(getMenuRecipes).toHaveBeenCalled;
    expect(addMenuRecipes).toHaveBeenCalled;
  });

});
