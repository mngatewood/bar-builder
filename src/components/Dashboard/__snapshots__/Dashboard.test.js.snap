// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    addMenuRecipes={[Function]}
    clearMenuRecipes={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/inventory"
>
          <button>
                    Update My Inventory
          </button>
</NavLink>,
        <button
          onClick={[Function]}
>
          View My Bar Menu
</button>,
      ],
      "className": "dashboard-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            Update My Inventory
</button>,
          "to": "/inventory",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Update My Inventory",
          },
          "ref": null,
          "rendered": "Update My Inventory",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View My Bar Menu",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "View My Bar Menu",
        "type": "button",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/inventory"
>
            <button>
                        Update My Inventory
            </button>
</NavLink>,
          <button
            onClick={[Function]}
>
            View My Bar Menu
</button>,
        ],
        "className": "dashboard-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              Update My Inventory
</button>,
            "to": "/inventory",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update My Inventory",
            },
            "ref": null,
            "rendered": "Update My Inventory",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View My Bar Menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "View My Bar Menu",
          "type": "button",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
